packages/atomic/src/components/facets-v1/atomic-rating-facet/atomic-rating-facet.tsx

```diff
@@ -115,8 +133,31 @@ export class AtomicRatingFacet
       .length;
   }
 
+  private generateCurrentValues() {
+    const currentValues: NumericRangeRequest[] = [];
+    for (let i = 0; i < this.numberOfStars; i++) {
+      currentValues.push(
+        buildNumericRange({
+          start: i,
+          end: i + 1,
+          endInclusive: true,
+        })
+      );
+    }
+    return currentValues;
+  }
+
+  private generateIconDisplay() {
+    const iconDisplay: VNode[] = [];
+    for (let i = 0; i < this.numberOfStars; i++) {
+      iconDisplay.push(this.icon);
+    }
+    return iconDisplay;
```

Comment:
This method is called once at the instantiation of the component. It displays the correct number of icons needed. I'm not clear on how we will be able to style the fill of individual stars if they will be the same HTML component with the same classes.